app/main.py

```diff
@@ -1,2 +1,80 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.length = 0
+        self.hash_table: list[Node | None] = [None] * self.capacity
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while ((self.hash_table[index] is not None)
+               and (self.hash_table[index].key != key)):
+
+            index = (index + 1) % self.capacity
+
+        return index
+
+    def __len__(self) -> int:
+        return self.length
+
+    @property
+    def _current_max_size(self) -> int:
+        return int(self.capacity * 2 / 3)
+
+    def _resize(self) -> None:
+        old_hash_table = self.hash_table
+        new_size = self.capacity * 2
+
+        self.__init__(new_size)
```

Comment:
The `_resize` method should not call `self.__init__(new_size)`, as this will reset the entire object, including its length and other attributes. Instead, update the `capacity` and create a new hash table of the new size, then rehash the existing elements into the new table.